LibreHardwareMonitorLib/Hardware/Storage/StorageGroup.cs

```diff
@@ -35,10 +39,98 @@ public StorageGroup(ISettings settings)
                 {
                     _hardware.Add(instance);
                 }
+
+                if (serialNumber != null && virtualDiskMap.ContainsKey(serialNumber))
+                {
+                    var physicalDisks = virtualDiskMap[serialNumber];
+                    foreach (PhysicalDisk physicalDisk in physicalDisks)
+                    {
+                        var physicialDiskIdx = Convert.ToUInt32(physicalDisk.deviceId);
+                        var physicalDiskInstance = AbstractStorage.CreateInstance($"\\\\.\\PHYSICALDRIVE{physicalDisk.deviceId}", physicialDiskIdx, physicalDisk.size, scsi, settings);
+                        if (physicalDiskInstance != null)
+                        {
+                            _hardware.Add(physicalDiskInstance);
+                        }
+                    }
+                }
             }
         }
     }
 
+    /// <summary>
+    /// Returns a map of serialNumber -> physical disk properties.
+    /// If you are using Windows Storage Spaces, one or more physical disks are put together to form a single disk (for better performance and/or resilience).
+    /// We can use the Storage Management API to find the device ids of these "hidden" physical disks. 
+    /// </summary>
+    /// <returns></returns>
+    private Dictionary<string, List<PhysicalDisk>> DiskToPhysicalDisk()
+    {
+        var physicalDiskMap = new Dictionary<string, PhysicalDisk>();
+        ManagementScope managementScope = new ManagementScope("\\\\.\\ROOT\\Microsoft\\Windows\\Storage");
+        //https://learn.microsoft.com/en-us/previous-versions/windows/desktop/stormgmt/msft-physicaldisk
+        using var physicalDiskSearcher = new ManagementObjectSearcher(managementScope, new ObjectQuery("SELECT * FROM MSFT_PhysicalDisk"));
+        foreach (ManagementBaseObject physicalDisk in physicalDiskSearcher.Get())
+        {
+            var sn = ExtractSerialNumber((string)physicalDisk["ObjectId"]);
+            var deviceId = (string)physicalDisk["DeviceId"];
+            var size = (ulong)physicalDisk["Size"];
+            if (sn != null)
+            {
+                physicalDiskMap.Add(sn, new PhysicalDisk(deviceId, size));
+            }
+        }
+
+        var virtualDiskMap = new Dictionary<string, List<PhysicalDisk>>();
```

Comment:
Might as well use a value tuple (string DeviceId, ulong Size) instead of the PhysicalDisk.